Sources/SwiftSyntaxBuilder/DeclBuildables.swift

```diff
@@ -0,0 +1,184 @@
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+
+// MARK: - Protocols
+
+public protocol DeclListBuildable: SyntaxListBuildable {
+  func buildDeclList(format: Format, leadingTrivia: Trivia) -> [DeclSyntax]
+}
+
+public protocol DeclBuildable: SyntaxBuildable, DeclListBuildable {
+  func buildDecl(format: Format, leadingTrivia: Trivia) -> DeclSyntax
+}
+
+extension DeclBuildable {
+  public func buildSyntax(format: Format, leadingTrivia: Trivia) -> Syntax {
+    buildDecl(format: format, leadingTrivia: leadingTrivia)
+  }
+
+  public func buildDeclList(format: Format, leadingTrivia: Trivia) -> [DeclSyntax] {
+    [buildDecl(format: format, leadingTrivia: leadingTrivia)]
+  }
+}
+
+// MARK: - Function Builder
+
+@_functionBuilder
+struct DeclListBuilder {
+  static func buildBlock(_ builders: DeclListBuildable...) -> DeclListBuildable {
+    DeclList(builders: builders)
+  }
+}
+
+// MARK: - List
+
+public struct DeclList: DeclListBuildable {
+  let builders: [DeclListBuildable]
+
+  public func buildDeclList(format: Format, leadingTrivia: Trivia) -> [DeclSyntax] {
+    builders.flatMap { $0.buildDeclList(format: format, leadingTrivia: leadingTrivia) }
+  }
+
+  public func buildSyntaxList(format: Format, leadingTrivia: Trivia) -> [Syntax] {
+    buildDeclList(format: format, leadingTrivia: leadingTrivia)
+  }
+}
+
+extension DeclList {
+  public static let empty: DeclList = DeclList(builders: [])
+}
+
+// MARK: - Buildables
+
+// MARK: Import
+
+public struct Import: DeclBuildable {
+  let moduleName: String
+
+  public init(_ moduleName: String) {
+    self.moduleName = moduleName
+  }
+
+  public func buildDecl(format: Format, leadingTrivia: Trivia) -> DeclSyntax {
+    let importToken = Tokens.import.withLeadingTrivia(leadingTrivia)
+    let moduleNameToken = SyntaxFactory.makeIdentifier(moduleName)
+
+    return ImportDeclSyntax {
+      $0.useImportTok(importToken)
+      $0.addPathComponent(AccessPathComponentSyntax {
+        $0.useName(moduleNameToken)
+      })
+    }
+  }
+}
+
+// MARK: Variables
+
+public protocol VariableMutability {
+  static var token: TokenSyntax { get }
+}
+
+public enum VariableLetMutability: VariableMutability {
+  public static let token = Tokens.let
+}
+
+public enum VariableVarMutability: VariableMutability {
+  public static let token = Tokens.var
+}
+
+public typealias Let = Variable<VariableLetMutability>
+public typealias Var = Variable<VariableVarMutability>
+
+public struct Variable<Mutability: VariableMutability>: DeclBuildable {
+  let name: String
+  let type: String
+  let initializer: ExprBuildable?
+
+  public init(_ name: String, of type: String, value: ExprBuildable? = nil) {
+    self.name = name
```

Comment:
I think `type` being `String` here is not great, but it's a good first step. I'd like to see more declarative wrappers around building types alongside declarations and expressions, and those APIs can eventually subsume this one.

Reply:
Agreed. In fact, I made a [`Type` type](https://github.com/akkyie/SyntaxBuilder/blob/master/Sources/SyntaxBuilder/Syntax/Type/Type.swift ) to express a type when it was just my toy, but couldn't figure it out how a straightforward API should look like.